node/src/app/cli.rs

```diff
@@ -173,8 +172,7 @@ impl Cli {
                 // initializer2_runner.run(&mut rng).await;
 
                 match initializer_runner.run(&mut rng).await {
-                    ReactorExit::ProcessShouldExit(ExitCode::Success) => return Ok(()),
-                    ReactorExit::ProcessShouldExit(exit_code) => process::exit(exit_code as i32),
```

Comment:
What was `process::exit` doing then? If we don't need it anymore.

Reply:
It exits the process immediately without running destructors.  This PR uses Marc's approach of passing the exit code out to `main` so we can drop everything and only then call `process::exit`.